spec/lang/types.md

```diff
@@ -178,6 +185,19 @@ impl Type {
 And we also define how to compute the actual size and alignment.
 
 ```rust
+impl TupleHeadLayout {
+    // This is one function to avoid quadratic computations.
```

Comment:
This comment isn't very clear.

Reply:
Actually I think I noticed a further-reaching problem, that since `compute_size` calls `tail.compute_size` and `tail.compute_align`, I think computing just the size might lead to exponentially (in the number of layers) many calls to the `compute_align` of the last tail.
So I will introduce a `LayoutStrategy::compute_size_align -> (Size, Align)` and have the `compute_size` be simply `compute_size_align.0`.

Reply:
Let's call it `compute_size_and_align`, that's also how this is called in rustc.